docker-compose.yml

```diff
@@ -0,0 +1,21 @@
+version: "3.9"
+services:
+  api:
+    build:
+      context: ./api
+      dockerfile: Dockerfile.dev
+    volumes:
+      - ./api:/home/app/api
```

Comment:
I am a bit confused about how volumes work here.

In the `Dockerfile.dev` we are `COPY`ing the whole `api` directory to `api` directory of the container. And then we are making it a volume. Won't those two conflict?

Reply:
No, making volume is just mapping both directories so that if any files changed at place the changes will also reflect at the mapped directory.

Reply:
Okay. Suppose you copy some other files to containers `api` directory. And then map your `api` directory to container's `api` directory. What happens in that case?

The files are different now.

Reply:
> In general, the Docker build command restricts the sources of files we can use in our Docker images. We
specify a build context, which is the root from which both the Dockerfile and all its dependent files must be
found.

I don't think we will be copying files from outside of the root of Dockerfile.

Reply:
I just want to understand what happens in that case. If it results in a conflict with different files, then the way we are using volume rn is not the way to do it.

Reply:
To simulate this I tried to build a image with a outer reference , got following error - 
```
=> ERROR [4/4] COPY /Users/brijsiyag/Desktop/docker-k8s/docker-django/manage.py .                                                               0.0s
------
 > [4/4] COPY /Users/brijsiyag/Desktop/docker-k8s/docker-django/manage.py .:
------
failed to compute cache key: "/Users/brijsiyag/Desktop/docker-k8s/docker-django/manage.py" not found: not found
```

Reply:
Cool, looks good.